File: Wox.UI.React/src/enums/WoxImageTypeEnum.ts
import {BaseEnum} from "./base/BaseEnum.ts";


export type WoxImageType = string

export class WoxImageTypeEnum extends BaseEnum {
    static readonly WoxImageTypeAbsolutePath = WoxImageTypeEnum.define("absolute", "absolute");
    static readonly WoxImageTypeRelativePath = WoxImageTypeEnum.define("relative", "relative");
    static readonly WoxImageTypeBase64 = WoxImageTypeEnum.define("base64", "base64");
    static readonly WoxImageTypeSvg = WoxImageTypeEnum.define("svg", "svg");
    static readonly WoxImageTypeUrl = WoxImageTypeEnum.define("url", "url");
}

File: Wox.UI.Flutter/wox/lib/enums/wox_image_type_enum.dart
typedef WoxImageType = String;

enum WoxImageTypeEnum {
  WOX_IMAGE_TYPE_ABSOLUTE_PATH("absolute", "absolute"),
  WOX_IMAGE_TYPE_RELATIVE_PATH("relative", "relative"),
  WOX_IMAGE_TYPE_BASE64("base64", "base64"),
  WOX_IMAGE_TYPE_SVG("svg", "svg"),
  WOX_IMAGE_TYPE_LOTTIE("lottie", "lottie"),
  WOX_IMAGE_TYPE_EMOJI("emoji", "emoji"),
  WOX_IMAGE_TYPE_URL("url", "url");

  final String code;
  final String value;

  const WoxImageTypeEnum(this.code, this.value);

  static String getValue(String code) => WoxImageTypeEnum.values.firstWhere((activity) => activity.code == code).value;
}
